core/services/ocrcommon/peer_wrapper.go

```diff
@@ -105,22 +105,64 @@ func (p *SingletonPeerWrapper) IsStarted() bool {
 // Start starts SingletonPeerWrapper.
 func (p *SingletonPeerWrapper) Start(context.Context) error {
 	return p.StartOnce("SingletonPeerWrapper", func() error {
-		// Peer wrapper panics if no p2p keys are present.
-		if ks, err := p.keyStore.P2P().GetAll(); err == nil && len(ks) == 0 {
-			return errors.Errorf("No P2P keys found in keystore. Peer wrapper will not be fully initialized")
-		}
-		key, err := p.keyStore.P2P().GetOrFirst(p.config.P2PPeerID())
+		peerConfig, err := p.PeerConfig()
 		if err != nil {
 			return err
 		}
-		p.PeerID = key.PeerID()
+
+		p.lggr.Debugw("Creating OCR/OCR2 Peer", "config", peerConfig)
+		// Note: creates and starts the peer
+		peer, err := ocrnetworking.NewPeer(peerConfig)
+		if err != nil {
+			return errors.Wrap(err, "error calling NewPeer")
+		}
+		p.Peer1 = &peerAdapterOCR1{
+			peer.OCR1BinaryNetworkEndpointFactory(),
+			peer.OCR1BootstrapperFactory(),
+		}
+		p.Peer2 = &peerAdapterOCR2{
+			peer.OCR2BinaryNetworkEndpointFactory(),
+			peer.OCR2BootstrapperFactory(),
+		}
+		p.peerCloser = peer
+		return nil
+	})
+}
+
+func (p *SingletonPeerWrapper) PeerConfig() (ocrnetworking.PeerConfig, error) {
```

Comment:
@jmank88 I'm not super happy with making this function public, but encountered a couple of problem with adding tests for this behaviour:
* The actual libocr.concretePeer doesn't seem to expose this information anywhere (it's all private), making it hard to assert the right thing has been set.
* Writing a package-only test (my first attempt) wouldn't compile since there's an import cycle with the `chainlink` package.

Therefore this seemed the easiest way to add tests for this behaviour, but please lmk if you have a suggestion :)

Reply:
Did you try a `package *_test`? That usually works around the cycle, but comes with some tradeoffs.

Reply:
Yep, that's what this is :) The trade-off is making PeerConfig() public

Reply:
We have a `helpers_test.go` pattern :shrug:

Reply:
Ah got it... I hadn't considered doing that. That does seem cleaner, so I'll go ahead with that option!